bridge_adaptivity/module/models.py

```diff
@@ -41,7 +40,20 @@ def _get_engine_driver(engine):
     return driver
 
 
-ENGINES = _discover_engines()
+def _get_grading_policy_cls(mod_name):
+    module = None
+    gp_module = importlib.import_module('module.policies.policy_{}'.format(mod_name))
+    for attr in inspect.getmembers(gp_module):
+        if attr[0].endswith('GradingPolicy'):
+            module = attr[1]
+    return module
+
+
+ENGINES = _discover_applicable_modules(folder_name='engines', file_startswith='engine_')
+
+GRADING_POLICY_MODULES = _discover_applicable_modules(folder_name='policies', file_startswith='policy_')
+GRADING_POLICY_NAME_TO_CLS = {name: _get_grading_policy_cls(name) for _, name in GRADING_POLICY_MODULES}
```

Comment:
I would try to avoid creating dict in the middle with combining GRADING_POLICY_CHOICES from the GRADING_POLICY_MODULES:
```
GRADING_POLICY_CHOICES = ((name, _get_grading_policy_cls(module).public_name) for module, name in GRADING_POLICY_MODULES)

Reply:
I use GRADING_POLICY_NAME_TO_CLS variable in GradingPolicy model.
```module/models.py:122```

Reply:
Got it, thanks.